api/src/stampy_chat/db/session.py

```diff
@@ -48,17 +48,15 @@ def __init__(self, engine=None, batch_size=100, save_every=1):
         self._last_save = time.time()
 
     def commit(self):
-        with Session(self.engine) as session:
-            try:
+        try:
+            with Session(self.engine) as session:
                 session.add_all(self.batch)
                 session.commit()
                 logger.debug('added %s items', len(self.batch))
-                self.batch = []
-            except SQLAlchemyError as e:
-                logger.warn('Got error when trying to commit to database: %s', e)
-                session.rollback()
-                raise e
+            self.batch = []
             self._last_save = time.time()
+        except SQLAlchemyError as e:
```

Comment:
there's an interesting Heisenbug with sessions being dropped - this won't fix it, but at least won't break the chat when it happens. It only happens when the chatbot isn't used for a long period of time, which hopefully shouldn't be an issue once things start being used?
An objection can be raised that this will result in interactions not getting logged, which is an issue, but not a pressing one, as this is only for logging purposes, and should only happen if the error occurs and then the worker is shut down - otherwise the interactions will be saved the next time something happens, as they're saved in batches anyway